gnina/training.py

```diff
@@ -661,31 +698,28 @@ def log_training_results(trainer):
         Evaluate metrics on the training set and update the LR according to the loss
         function, if needed.
         """
-        evaluator.run(train_loader)
+        train_evaluator.run(train_loader)
 
         for outstream in outstreams:
             utils.log_print(
-                evaluator.state.metrics,
+                train_evaluator.state.metrics,
                 title="Train Results",
                 epoch=trainer.state.epoch,
                 epoch_time=trainer.state.times["EPOCH_COMPLETED"],
                 elapsed_time=elapsed_time.total,
                 stream=outstream,
             )
 
-        mts = evaluator.state.metrics
+        mts = train_evaluator.state.metrics
         metrics_train["Epoch"].append(trainer.state.epoch)
         for key, value in mts.items():
             metrics_train[key].append(value)
 
         # Update LR based on the loss on the training set
         if args.lr_dynamic:
-            loss = mts["Loss (pose)"]
-            try:
-                loss += mts["Loss (affinity)"]
-            except KeyError:
-                # No affinity loss
-                pass
```

Comment:
This `try/except` block was very dangerous. Changing the name of the metric without changing the key here would have failed silently.